### PR TITLE
Add impeller key to skia gold client, Turn on a framework test shard that will run unit tests with --enable-impeller

### PR DESCRIPTION
**Pull request by jonahwilliams**
Adds a Skia Gold key that describes which Impeller renderer, if any is running, trying to leave the existing hashes unaltered.

Turns on a test shard that will run impeller on macOS, though this won't run without the PR being submitted first.

### CODE REVIEW

File: .ci.yaml [hunk 1]
```diff
@@ -3335,6 +3335,33 @@ targets:
       - bin/**
       - .ci.yaml
 
+  - name: Mac framework_tests_impeller
+    recipe: flutter/flutter_drone
+    timeout: 60
+    bringup: true
+    properties:
+      cpu: x86 # https://github.com/flutter/flutter/issues/119880
```
Comment (christopherfujino) [inline .ci.yaml:3343]: should the name just be `Mac_arm64 framework_tests_impeller` instead? cc @godofredoc

File: dev/bots/test.dart [hunk 2]
```diff
@@ -815,6 +815,16 @@ Future<void> _runFrameworkTests() async {
     );
   }
 
+  Future<void> runImpeller() async {
+    printProgress('${green}Running packages/flutter tests $reset in Impeller$reset');
+    // Try compiling code outside of the packages/flutter directory with and without --track-widget-creation
```
Comment (christopherfujino) [inline dev/bots/test.dart:820]:
```suggestion
```

File: dev/integration_tests/flutter_gallery/macos/Podfile.lock [hunk 3]
```diff
@@ -23,13 +27,16 @@ EXTERNAL SOURCES:
...
   connectivity_macos: 5dae6ee11d320fac7c05f0d08bd08fc32b5514d9
   FlutterMacOS: 8f6f14fa908a6fb3fba0cd85dbd81ec4b251fb24
   Reachability: 33e18b67625424e47b6cde6d202dce689ad7af96
   url_launcher_macos: d2691c7dd33ed713bf3544850a623080ec693d95
+  video_player_avfoundation: e9e6f9cae7d7a6d9b43519b0aab382bca60fcfd1
 
 PODFILE CHECKSUM: 2e6060c123c393d6beb3ee5b7beaf789de4d2e47
 
-COCOAPODS: 1.12.1
+COCOAPODS: 1.13.0
```
Comment (christopherfujino) [inline dev/integration_tests/flutter_gallery/macos/Podfile.lock:42]: I'm guessing revert this in case cocoapods on CI is older?

Comment (jonahwilliams) [reply dev/integration_tests/flutter_gallery/macos/Podfile.lock:42]: Agh, yeah I didn't notice that - thank you

**Review by christopherfujino — APPROVED** [review]
LGTM

**Review by Piinks — APPROVED** [review]
LGTM!

**Comment by jonahwilliams** [comment]
Ahh, @christopherfujino when I accepted your suggestion it added a co-authored commit with your gmail address, which doesn't have a google CLA 🤣

**Comment by jonahwilliams** [comment]
Redo here due to CLA issues: https://github.com/flutter/flutter/pull/141341

**Comment by christopherfujino** [comment]
> Ahh, @christopherfujino when I accepted your suggestion it added a co-authored commit with your gmail address, which doesn't have a google CLA 🤣

Yeah sorry. This is a bug in the clabot, it knows about both emails, but only when i'm the only author on the PR, apparently.

**Pull request closed without merging**